src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -141,6 +143,22 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         filteredPersons.setPredicate(predicate);
     }
 
+    @Override
+    public void batchDeleteWithPredicate(Predicate<Person> predicate) {
+        requireNonNull(predicate);
+        updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        filteredPersons.setPredicate(predicate);
+        ObservableList<Person> listToDelete = this.getFilteredPersonList();
+
+        List<Person> copyList = new ArrayList<>(listToDelete);
+        for (Person person : copyList) {
+            addressBook.removePerson(person);
+        }
+
+        updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
```

Comment:
So this is the point after performing the batch delete, you update the FilteredList again.